### PR TITLE
Add simple tests and adjust our workspace internal module resoultion logic

### PR DESCRIPTION
**Pull request by ggilmore**
This PR includes a simple test project with no third party dependencies, along with a corresponding test suite. It's not exhaustive (since there are no external dependencies), but having test cases that don't break due to not having pinned versions of external packages is valuable. 

----------------

In the course of writing these tests, I discovered an issue when we try to jump to def / hover over this line: https://github.com/sourcegraph/python-langserver/blob/simple-tests/test/repos/fizzbuzz_service/fizzbuzz_service/checkers/fizzbuzz/fizzbuzz_checker.py#L6:25 (No definition would be returned).

After tracing our modifications to jedi, I realized that our logic to search for internal project packages seems incorrect. 

The relevant directory structure for the file mentioned above is:

```
/fizzbuzz_service
/fizzbuzz_service/checkers
/fizzbuzz_service/checkers/fizz
/fizzbuzz_service/checkers/fizzbuzz
/fizzbuzz_service/checkers/fizzbuzz/fizzbuzz_checker.py
```

`fizzbuzz_checker.py` has a relative import that the top of the file for `..fizz`.
The `..` refers to the parent `checkers` package. 

Previously, when resolving the `checkers` package (and weren't given any specific paths to check for `checkers` from jedi) - we'd only consider the paths:

- `/checkers`
- `/fizzbuzz_service/checkers/fizzbuzz`,
- `/fizzbuzz_service/checkers/fizzbuzz/checkers` 

when looking for the `checkers` package. None of the paths above are correct (and two don't exist). We'd then assume that `checkers` is some external `pypi` package. `checkers` isn't a `pypi` package, so fetching that would fail and cause the response to be incorrect. 

The fix is to more closely mimic jedi's strategy when resolving these imports. 
When jedi is asked to resolve the `checkers` import on `/fizzbuzz_service/checkers/fizzbuzz/fizzbuzz_checker.py`, it:

- walks up all the parents of `/fizzbuzz_service/checkers/fizzbuzz/fizzbuzz_checker.py`:

   - `/fizzbuzz_service/checkers/fizzbuzz`
   - `/fizzbuzz_service/checkers`
   - `/fizzbuzz_service`
    - `/`

- if any of those parents end in a folder named `checkers`, jedi saves them, 
e.g. `matching_dirs = ['/fizzbuzz_service/checkers']`
- and then it inserts `matching_dirs` to the start of `sys.path`, which is used during python's native module resolution strategy. 

My solution mimics this behavior, and includes a test case for the above example.

### CODE REVIEW

File: langserver/jedi.py [hunk 1]
```diff
@@ -63,11 +63,7 @@ def find_module_remote(string, dir=None, fullname=None):
             """A swap-in replacement for Jedi's find module function that uses the
             remote fs to resolve module imports."""
             if dir is None:
-                # If we're starting the search for a module, then proceed from both the project root as well as the
-                # folder containing the current file. Normally we should only search from the project root (I think),
-                # but certain repos such as collections of example projects are structured such that we can support
-                # them better by searching relative to the current file, too.
-                dir = ["/", os.path.dirname(path)]
+                dir = get_module_search_paths(string, path)
```
Comment (attfarhan) [inline langserver/jedi.py]: @attfarhan

**Comment by ggilmore** [comment]
I'd appreciate someone to double check the logic behind my fix, and to make sure that it's correct behavior and mimics what python/jedi actually does.

**Comment by sqs** [comment]
> I'd appreciate someone to double check the logic behind my fix, and to make sure that it's correct behavior and mimics what python/jedi actually does.

Reviewing now

File: langserver/jedi.py [hunk 2]
```diff
@@ -140,3 +136,29 @@ def load_source(path) -> str:
...
+    '''
+    Provides an ordered list of directories in the workspace to search for the 
+    given 'module_name', starting from the directory that the script is operating on. 
+    
+    This mimics Jedi's modifications of sys.path that it uses during module resolution.
+    See: 
+    https://sourcegraph.com/github.com/sourcegraph/jedi/-/blob/jedi/evaluate/imports.py#L237:9
+    '''
+    for parent in traverse_parents(script_file_path):
+        if os.path.basename(parent) == module_name:
```
Comment (sqs) [inline langserver/jedi.py]: This is correct for 99.99% of the cases, but I recommend adding a note that there are some edge cases for pull PEP-0235 compatibility with respect to case (https://www.python.org/dev/peps/pep-0235/). Definitely not worth fixing now, though!

**Pull request merged**